afs/afs-core/src/main/java/com/powsybl/afs/OrderedDependencyManager.java

```diff
@@ -0,0 +1,96 @@
+package com.powsybl.afs;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class OrderedDependencyManager {
+
+    private ProjectFile projectFile;
+
+    private List<ProjectDependency<ProjectNode>> dependencyCache = null;
+
+    public OrderedDependencyManager(ProjectFile projectFile) {
+        this.projectFile = projectFile;
+        projectFile.addListener(new DefaultProjectFileListener() {
+            @Override
+            public void dependencyChanged(String name) {
+                dependencyCache = null;
+            }
+        });
+    }
+
+    public void appendDependencies(String name, List<ProjectNode> projectNodes) {
+        List<ProjectNode> nodes = getDependencies(name);
+        nodes.addAll(projectNodes);
+        setDependencies(name, nodes);
+    }
+
+    public void insertDependencies(String name, int index, List<ProjectNode> projectNodes) {
+        List<ProjectNode> nodes = getDependencies(name);
+        nodes.addAll(index, projectNodes);
+        setDependencies(name, nodes);
+    }
+
+    public void removeDependency(String name, int index) {
+        List<ProjectNode> nodes = getDependencies(name);
+        nodes.remove(index);
+        setDependencies(name, nodes);
+    }
+
+    public void removeDependencies(String name, List<String> nodeIds) {
+        List<ProjectNode> nodes = getDependencies(name);
+        setDependencies(name, nodes.stream().filter(dep -> !nodeIds.contains(dep.getId())).collect(Collectors.toList()));
+    }
+
+    public void setDependencies(String name, List<ProjectNode> projectNodes) {
+        getDependenciesFor(name).map(el -> el.getRight().getName()).forEach(projectFile::removeDependencies);
```

Comment:
Is this really efficient? Could we remove several dependencies at the same time?

Reply:
no it isn't.
But it rely on the only available method (`void removeDependency(String nodeId, String name, String toNodeId)`) on `AppStorage`.
Making it more effecient by batching the call would require changing the interface and implement this on all AppStorage.

Reply:
It is very unlikely we have performance issue with this method as there is always a few numbers of dependency.